### PR TITLE
Upgrade to Rubocop 0.57.2

### PR DESCRIPTION
**Pull request by apjanke**
- [X] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/CONTRIBUTING.md) document?
- [X] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [X] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/homebrew/pull/49031). *N/A. Brew style rubocop behavior is not tested; I don't think.*
- [X] Have you successfully run `brew style` with your changes locally?
- [X] Have you successfully run `brew tests` with your changes locally?

-----

Any interest in upgrading to Rubocop 0.57.1? I thought it might fix an error I was running in to with `brew style --rspec`. It didn't, but I went and did the work for it anyway (including style fixes) before realizing that. Do we care about being on a newer version of Rubocop?

### CODE REVIEW

**Comment by ilovezfs** [comment]
I'd like to see the `EOS is not aligned with` thing whitelisted, at least for core, because it would affect the caveats and plist formatting unnecessarily and rubocop seems to invent some new heredoc pedantry every version.

**Comment by MikeMcQuaid** [comment]
@reitermarkus `rubocop-cask` needs updated here, thanks. Would be good maybe do handle merging that into Homebrew/brew next on the GSoC after search?

@apjanke will need a rebase.

> I'd like to see the EOS is not aligned with thing whitelisted, at least for core, because it would affect the caveats and plist formatting unnecessarily and rubocop seems to invent some new heredoc pedantry every version.

I disagree here; the changes that have been made here seem to be more consistent and readable (and more like other Ruby blocks) in their indentation style. I do think we should do the usual "fix Homebrew/homebrew-core styles" PR before this is merged and discuss things there.

**Comment by MikeMcQuaid** [comment]
> I do think we should do the usual "fix Homebrew/homebrew-core styles" PR before this is merged and discuss things there.

Have opened https://github.com/Homebrew/homebrew-core/pull/28754 to discuss.

**Comment by MikeMcQuaid** [comment]
@apjanke Any chance of a rebase here? Thanks!

**Comment by apjanke** [comment]
Rebased!

File: Library/Homebrew/cask/lib/hbc/locations.rb [hunk 1]
```diff
@@ -0,0 +1,31 @@
+module Hbc
```
Comment (reitermarkus) [inline Library/Homebrew/cask/lib/hbc/locations.rb]: This file doesn't exist anymore in `master`.

File: Library/Homebrew/constants.rb [hunk 2]
```diff
@@ -1,5 +1,5 @@
 # frozen_string_literal: true
 
 # RuboCop version used for `brew style` and `brew cask style`
-HOMEBREW_RUBOCOP_VERSION = "0.55.0"
+HOMEBREW_RUBOCOP_VERSION = "0.57.1"
 HOMEBREW_RUBOCOP_CASK_VERSION = "~> 0.19.0" # has to be updated when RuboCop version changes
```
Comment (reitermarkus) [inline Library/Homebrew/constants.rb]: Bump to `0.20.0`.

**Comment by apjanke** [comment]
Amended to address @reitermarkus's comments.

File: Library/Homebrew/constants.rb [hunk 3]
```diff
@@ -1,5 +1,5 @@
 # frozen_string_literal: true
 
 # RuboCop version used for `brew style` and `brew cask style`
-HOMEBREW_RUBOCOP_VERSION = "0.55.0"
-HOMEBREW_RUBOCOP_CASK_VERSION = "~> 0.19.0" # has to be updated when RuboCop version changes
+HOMEBREW_RUBOCOP_VERSION = "0.57.1"
```
Comment (reitermarkus) [inline Library/Homebrew/constants.rb]: `0.57.2`

Comment (apjanke) [reply Library/Homebrew/constants.rb]: Amended.

File: Library/Homebrew/.rubocop.yml [hunk 4]
```diff
@@ -72,6 +77,11 @@ Naming/PredicateName:
 Naming/UncommunicativeMethodParamName:
   Enabled: false
 
+# I'm not sure how to correct these, and seems to get false positives on
+# modifiers used on symbols of methods
```
Comment (reitermarkus) [inline Library/Homebrew/.rubocop.yml]: Change this comment to reference https://github.com/rubocop-hq/rubocop/issues/5953.

Comment (apjanke) [reply Library/Homebrew/.rubocop.yml]: Done.

**Comment by MikeMcQuaid** [comment]
Thanks @apjanke!

**Comment by apjanke** [comment]
Yay!

**Pull request merged**